src/scheduler.cc

```diff
@@ -267,7 +267,47 @@ Status SchedulerService::TaskInfo(ServerContext* context, const TaskInfoRequest*
   return Status::OK;
 }
 
-void SchedulerService::deliver_object_if_necessary(ObjRef canonical_objref, ObjStoreId from, ObjStoreId to) {
+Status SchedulerService::KillWorkers(ServerContext* context, const KillWorkersRequest* request, KillWorkersReply* reply) {
+  // TODO: Update reference counts
+  std::lock_guard<std::mutex>
```

Comment:
I have but I don't see the benefit over acquiring the locks I actually need. I would only use that method for debugging or other "side" purposes like inspecting/dumping the scheduler state, not for any method that already needs exactly what it needs access to.